website/blog/2023-03-10-databend-weekly-84.md

```diff
@@ -0,0 +1,178 @@
+
+We fixed some clippy warnings that we encountered. One notable thing is that we now use `#[default]` to annotate the default value of an `enum`, as shown below.
+
+```rust
+#[derive(Debug, Default)]
+pub enum RecordDelimiter {
+    #[default]
+    Crlf,
+    Any(u8),
+}
+```
+
+If you are interested in this Rust trick, you can read this RFC: [`derive_default_enum`](https://rust-lang.github.io/rfcs/3107-derive-default-enum.html).
+
+### Announcing OpenDAL v0.30
+
+[OpenDAL](https://github.com/datafuselabs/opendal) brings some important changes in its newly released 0.30 version, including removing `Object` Abstraction and bindings for JavaScript and Python.
+
+**Removing `Object` Abstraction**
+
+The term `Object` is applied in various fields, making it difficult to provide a concise definition of `opendal::Object`. We eliminated the intermediate API layer of `Object` and enabled users to directly utilize `Operator`. For example:
+
+```diff
+# get metadata of a path
+- op.object("path").stat().await;
++ op.stat("path").await;
+```
+
+**Bindings for JavaScript and Python**
+
+With [@suyanhanx](https://github.com/suyanhanx) [@messense](https://github.com/messense) [@Xuanwo](https://github.com/Xuanwo) and others' help, we released opendal's JavaScript and Python bindings. Now users of these languages can access data in various services with opendal. This is useful for Data Scientist and Data Analyst who **only** need opendal and **no** other SDKs.
+
+Here is a Python example:
+
+```python
+>>> import opendal
+>>> op = opendal.Operator('memory')
+>>> op.write("test", b"Hello, World!")
+>>> op.read("test")
+b'Hello, World!'
+```
+
+## What's Up Next
+
+We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.
+
+### Support More Mathematical Operators like PostgreSQL
+
+More mathematical operators in SQL expressions are wanted to perform complex calculations on data without having to write them in another language or use external tools.
+
+| Operator | Description         | Example    | Result |
+| -------- | ------------------- | ---------- | ------ |
+| ^        | exponentiation      | 2.0 ^ 3.0  | 8      |
+| \|/      | square root         | \|/ 25.0   | 5      |
+| \|\|/    | cube root           | \|\|/ 27.0 | 3      |
+| !        | factorial           | 5 !        | 120    |
+| @        | absolute value      | @ -5.0     | 5      |
+| &        | bitwise AND         | 91 & 15    | 11     |
+| \|       | bitwise OR          | 32 \| 3    | 35     |
+| #        | bitwise XOR         | 17 # 5     | 20     |
+| ~        | bitwise NOT         | ~1         | -2     |
+| <<       | bitwise shift left  | 1 << 4     | 16     |
+| >>       | bitwise shift right | 8 >> 2     | 2      |
+
+[@jun0315](https://github.com/jun0315) is working on this issue and has made some great progress. If you are interested in this feature, you are welcome to join the contribution.
```

Comment:
[@jun0315](https://github.com/jun0315) is working on it and has made some great progress. If you are interested in this feature, feel free to drop in and contribute.